plugins/CoreAdminHome/Commands/SetConfig/ConfigSettingManipulation.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\CoreAdminHome\Commands\SetConfig;
+
+use Piwik\Config;
+
+/**
+ * Representation of a INI config manipulation operation. Only supports two types
+ * of manipulations: appending to a config array and assigning a config value.
+ */
+class ConfigSettingManipulation
+{
+    /**
+     * @var string
+     */
+    private $sectionName;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string
+     */
+    private $value;
+
+    /**
+     * @var bool
+     */
+    private $isArrayAppend;
+
+    /**
+     * @param string $sectionName
+     * @param string $name
+     * @param string $value
+     * @param bool $isArrayAppend
+     */
+    public function __construct($sectionName, $name, $value, $isArrayAppend = false)
+    {
+        $this->sectionName = $sectionName;
+        $this->name = $name;
+        $this->value = $value;
+        $this->isArrayAppend = $isArrayAppend;
+    }
+
+    /**
+     * Performs the INI config manipulation.
+     *
+     * @param Config $config
+     * @throws \Exception if trying to append to a non-array setting value or if trying to set an
+     *                    array value to a non-array setting
+     */
+    public function manipulate(Config $config)
+    {
+        $setting =& $this->getSettingToManipulate($config);
```

Comment:
> References are the problem, not OOP

Sure, but all objects are passed and stored by reference. Any variable that is set to an object is a reference to the actual data. So you can't say references are bad w/o saying OOP is bad. In PHP, this specific feature isn't so much "references" but "references to POD types". I don't see a real problem w/ being able to store a reference to int/string/float. This functionality is available in other languages, though it may look and feel better when used in those languages than in PHP.